composer.json

```diff
@@ -10,22 +10,35 @@
         {"name": "Roman Borschel", "email": "roman@code-factory.org"},
         {"name": "Benjamin Eberlei", "email": "kontakt@beberlei.de"},
         {"name": "Jonathan Wage", "email": "jonwage@gmail.com"},
-        {"name": "Johannes Schmitt", "email": "schmittjoh@gmail.com"}
+        {"name": "Johannes Schmitt", "email": "schmittjoh@gmail.com"},
+        {"name": "Fabio B. Silva", "email": "fabio.bat.silva@gmail.com"}
     ],
     "require": {
-        "php": ">=5.3.2",
-        "doctrine/lexer": "1.*"
+        "php":          ">=7.0.0",
+        "hoa/compiler": "3.*"
     },
     "require-dev": {
-        "doctrine/cache": "1.*",
-        "phpunit/phpunit": "4.*"
+        "doctrine/cache"                        : "1.*",
+        "phpunit/phpunit"                       : "5.*",
+        "instaclick/coding-standard"            : "~1.1",
+        "squizlabs/php_codesniffer"             : "~1.5",
+        "satooshi/php-coveralls"                : "~1.0",
+        "instaclick/object-calisthenics-sniffs" : "dev-master",
+        "instaclick/symfony2-coding-standard"   : "dev-remaster"
     },
     "autoload": {
-        "psr-0": { "Doctrine\\Common\\Annotations\\": "lib/" }
+        "psr-4": {
+            "Doctrine\\Annotations\\": "src/"
```

Comment:
Indeed. But breaking BC just because you can makes it harder for anyone using your library, and means that they will take much more time to upgrade to the newer version (and ask you to continue providing support for the older version in parallel for much longer).

Rewriting all internals of the library to use different dependencies may be worth it (but we still wait for benefits/goals asked by @schmittjoh), and breaking BC to use PHP 7 return typehints is something I can agree with (it will make it impossible for people to support for 1.x and 2.x if they are extending Doctrine classes, but they should not need to use inheritance on them IMO anyway). But being friendlier to consumers of the library would be good.
Due to the PHP 7 requirement, I can guarantee you that Symfony cannot switch to the v2 only until Symfony 4 (as Symfony 3.x still supports PHP 5.x). This gives us 2 options:
- supporting 1.x only
- supporting 1.x and 2.x at the same time (letting people using PHP 7 upgrade to doctrine/annotations 2.x in their projects)

The second case is better (as the version supported by Symfony also impacts all projects build on top of Symfony, and their compatibility with other packages using doctrine/annotations), but only if we can easily use both. Currently the only changes in the Reader interface are its namespace and scalar typehints. adding scalar typehints does not hurt us (as long as we always call it properly by passing a string, the scalar typehint change does not impact the caller. The signature change only impacts people wanting to implement the interface). the namespace change however makes it a huge pain, as we don't have an interface anymore, but 2 different interfaces, and so we cannot typehint anymore when supporting both.
I don't see any benefit in the class renaming, and only drawbacks here (btw, this is why Twig still does not use namespaces for instance, and won't use them either in 2.x: the BC break would split the community using it without bringing any real benefit)